managed/src/main/java/com/yugabyte/yw/controllers/EncryptionAtRestController.java

```diff
@@ -42,6 +48,32 @@
 
   @Inject Commissioner commissioner;
 
+  @Inject CloudAPI.Factory cloudAPIFactory;
+
+  private void validateKMSProviderConfigFormData(ObjectNode formData, String keyProvider) {
+    if (keyProvider.toUpperCase().equals(KeyProvider.AWS.toString())
+        && (formData.get("AWS_ACCESS_KEY_ID") != null
+            || formData.get("AWS_SECRET_ACCESS_KEY") != null)) {
+      CloudAPI cloudAPI = cloudAPIFactory.get(KeyProvider.AWS.toString().toLowerCase());
+      Map<String, String> config = new HashMap<String, String>();
+      config.put("AWS_ACCESS_KEY_ID", formData.get("AWS_ACCESS_KEY_ID").textValue());
+      config.put("AWS_SECRET_ACCESS_KEY", formData.get("AWS_SECRET_ACCESS_KEY").textValue());
+      if (cloudAPI != null
+          && !cloudAPI.isValidCreds(config, formData.get("AWS_REGION").textValue())) {
+        throw new YWServiceException(BAD_REQUEST, "Invalid AWS Credentials.");
+      }
+    }
+    if (keyProvider.toUpperCase().equals(KeyProvider.SMARTKEY.toString())) {
+      final Set<String> apiUrl =
```

Comment:
Why final + why not simply:
`
List<String> apiUrl = Arrays.asList(...);
`

Reply:
Also why are we limited by these three domains only? Maybe it is better to use a mask like api.*.smartkey.io?

Reply:
@SergeyPotachev those are the only valid APIs that smartkey supports. From their documentation: `Note that the U.S. API endpoint is api.amer.smartkey.io, the E.U API endpoint is api.eu.smartkey.io, and the U.K. API endpoint is api.uk.smartkey.io`

Reply:
@daniel-yb OK, I see.